internal/output/output.go

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+// Package output manages outputting text to the user.
+//
+// Trace("Something very low level.") - not localized
+// Debug("Useful debugging information.") - not localized
+// Info("Something noteworthy happened!") - localized
+// Warn("You should probably take a look at this.") - localized
+// Error("Something failed but I'm not quitting.") - localized
+// Fatal("Bye.") - localized
+//
+//	calls os.Exit(1) after logging
+//
+// Panic("I'm bailing.") - not localized
+//
+//	calls panic() after logging
+package output
+
+import (
+	"fmt"
+	. "github.com/microsoft/go-sqlcmd/internal/output/formatter"
+	"github.com/microsoft/go-sqlcmd/internal/output/verbosity"
+	"github.com/microsoft/go-sqlcmd/pkg/sqlcmd"
+	"github.com/pkg/errors"
+	"io"
+	"regexp"
+	"strings"
+)
+
+var formatter Formatter
+var loggingLevel verbosity.Enum
+var runningUnitTests bool
+
+var standardWriteCloser io.WriteCloser
```

Comment:
>andardWriteCloser io.WriteCloser [](http://example.com/codeflow?start=6&length=32)

don't we need separate error and stdout channels? Most apps don't write everything to stdout, they use stderr too.

Legacy Sqlcmd also lets the user change where stdout and stderr write during app execution. Do you expect we'll keep that functionality with the new API such that this channel needs to support dynamic reassignment?

Reply:
I started with two channels, but then didn't know what semantics to apply to do the separation, so I removed stderr to keep this PR contained.  I've added an Issue, and assigned to myself:

[https://github.com/microsoft/go-sqlcmd/issues/169](https://github.com/microsoft/go-sqlcmd/issues/169)